Issue 561: Rename variables to reflect that they are choice maps

SamuelBrand1 opened the pull request:
This PR closes #561 .

This PR renames instances where a variable is a `ChoiceMap` to use `choice_` prefix consistently, i.e. use 'choice_value' and 'choice_gradient' instead of 'value_trie' and 'gradient_trie' for clarity and alignment with choicemap terminology. I've left variables that are typed to be `Trie`s.

Main danger is that my understanding of the static IR for Gen is that the `gensym`s for e.g value_trie are also meant to reflect creating a local variable name for a choice map object. If this is incorrect then I should revert those changes.

SamuelBrand1 committed:
Rename value_trie and gradient_trie to choice_value and choice_gradient

Refactored code and tests to consistently use 'choice_value' and 'choice_gradient' instead of 'value_trie' and 'gradient_trie' for clarity and alignment with choicemap terminology. Updated variable names, function signatures, and related comments across inference, static IR, and test files.

ztangent commented:
Thanks for this! Everything seems to pass, but could we change `choice_gradient` to `choice_grads`? I'm a little worried that `choice_gradient` is a little too close to the name of the `choice_gradients` function itself. `choice_grads` is also what is used within the documentation of the `choice_gradients` function.

Your understanding of the static IR is correct!

SamuelBrand1 commented:
> Thanks for this! Everything seems to pass, but could we change `choice_gradient` to `choice_grads`? I'm a little worried that `choice_gradient` is a little too close to the name of the `choice_gradients` function itself. `choice_grads` is also what is used within the documentation of the `choice_gradients` function.
> 
> Your understanding of the static IR is correct!

Thats a good point, I'll make a quick change.

SamuelBrand1 commented:
Having another look at the code base I found

https://github.com/probcomp/Gen.jl/blob/8f2f4cc0b8da19c9068d3bae889923248fd0f6b0/src/dynamic/backprop.jl#L349-L350

this has the other way of representing the variables e.g. `gradient_choices` etc. I think it would be nice to lock onto one convention to make the src code easier to read?

ztangent commented:
Yes, feel free to change that as well!

SamuelBrand1 committed:
Gone for `_choices`

Renamed 'values_trie' and 'gradient_trie' to 'value_choices' and 'gradient_choices' in HMC and MAP optimization code, as well as corresponding test cases, to improve code readability and clarity.

SamuelBrand1 committed:
More consistency on arg refactor

ztangent reviewed (changes requested):
Looks good but let's update the docstring and the static IR code as well.

SamuelBrand1 committed:
value choices refactor

SamuelBrand1 committed:
to gradient_choices

SamuelBrand1 committed:
to value_choices

SamuelBrand1 committed:
to gradient_choices

SamuelBrand1 committed:
to gradient_choices

SamuelBrand1 committed:
to gradient_choices

SamuelBrand1 committed:
rename funcs

SamuelBrand1 committed:
Rename variables in choice_gradients test

SamuelBrand1 committed:
Rename variables in choice_gradients docstring

ztangent reviewed (approved):
Looks good to me, thank you!
